test/test_jit.py

```diff
@@ -2874,6 +2888,27 @@ def checkScript(self,
 
         return ge
 
+    def test_annoying_doubles(self):
+        mod = types.ModuleType("temp")
+        mod.inf = float("inf")
+        mod.ninf = float("-inf")
+        mod.nan = float("nan")
+
+        with self.disableModuleHook():
+            @torch.jit.script
+            def foo():
+                return math.pi, 0.1, mod.inf, mod.ninf, 2.225073858507201e-308, mod.nan
```

Comment:
How do you get `math.pi` work without having pyobj cast and no declarations like `mod` outside?

Reply:
Sorry my fault, looks like we still use `PythonModuleValue` when creating the IR. Notice that there looks to be a casting bug in the `toSugaredValue` in [here](https://github.com/pytorch/pytorch/blob/b807970aeaf3bd26068f347a01196b66c0a21bb5/torch/csrc/jit/script/init.cpp#L281), where we are casting the pyobj to float. It should be double since our IValue is using double only, otherwise we will lose precision and things like `math.pi` won't be the correct value.